### PR TITLE
Remove go-mruby vendor  hacks

### PR DESCRIPTION
**Pull request by sponomarev**
Motivation

0. Moving to go modules in future
1. We don't need them anymore since all platform-specific builds are already held on specific platforms:  Linux/OSX
2. In any case `dep ensure` ~fixes~ rewrites these hacks during the build.

P.S. I've noticed  specific mruby build tuning with `MRUBY_CROSS_OS`. This configuration is not used for builds atm. Is it not something we forgot to add to CircleCI config?

### CODE REVIEW

**Comment by palkan** [comment]
> I've noticed specific mruby build tuning with `MRUBY_CROSS_OS`. This configuration is not used for builds atm. Is it not something we forgot to add to CircleCI config?

That was used to cross-compile mruby locally; we don't need it anymore

File: Makefile [hunk 1]
```diff
@@ -25,13 +25,15 @@ install-with-mruby:
 	go install -tags mrb ./...
 
 build:
-	env go build -tags mrb -ldflags $(LD_FLAGS) -o $(OUTPUT) cmd/anycable-go/main.go
+	go build -tags mrb -ldflags $(LD_FLAGS) -o $(OUTPUT) cmd/anycable-go/main.go
 
 prepare-cross-mruby:
```
Comment (palkan) [inline Makefile]: So, let's get rid of this too.

And here too: https://github.com/anycable/anycable-go/blob/master/etc/build_config.rb

File: Makefile [hunk 2]
```diff
@@ -25,13 +25,11 @@ install-with-mruby:
...
 build:
-	env go build -tags mrb -ldflags $(LD_FLAGS) -o $(OUTPUT) cmd/anycable-go/main.go
-
-prepare-cross-mruby:
-	(cd vendor/github.com/mitchellh/go-mruby && MRUBY_CROSS_OS=linux MRUBY_CONFIG=../../../../../../etc/build_config.rb make libmruby.a)
+	go build -tags mrb -ldflags $(LD_FLAGS) -o $(OUTPUT) cmd/anycable-go/main.go
 
 prepare-mruby:
-	(cd vendor/github.com/mitchellh/go-mruby && MRUBY_CONFIG=../../../../../../etc/build_config.rb make libmruby.a)
+	cd vendor/github.com/mitchellh/go-mruby && \
```
Comment (sponomarev) [inline Makefile]: **Shell Scripting 201**

Just to explain why it was changed: less confusion and more readability. 
Why can we do this: each make command is performed in a separate shell session
Why can't we write it with just separate lines: each line in command is performed in a separate shell session

Outside of a makefile, e.g. on CI, `pushd` and `popd` are usually used.

**Comment by sponomarev** [comment]
Once it's merged we can:
- upgrade go-mruby (it's already fixed without your PR, sorry). I've already checked that.
- provide a developer-friendly docker environment. In any case, I need it to debug #30

**Pull request merged**